client/src/components/TestAuth/index.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext } from 'react';
+import { context } from '../userProvider';
+
+function TestAuth() {
+  const [userData, setUserData] = useContext(context);
+  const handleLogin = async () => {
+    const req = await fetch('https://jsonplaceholder.typicode.com/posts/1');
+    await req.json();
+    setUserData({ ...userData, isAuthenticated: true });
+  };
+  const handleLogout = async () => {
+    const req = await fetch('https://jsonplaceholder.typicode.com/posts/1');
+    await req.json();
+    setUserData({ ...userData, isAuthenticated: false });
+  };
+  return (
+    <div>
+      <button type="button" onClick={handleLogin}>
+        Login
+      </button>
+      <button type="button" onClick={handleLogout}>
+        Logout
+      </button>
+    </div>
+  );
+}
+export default TestAuth;
+
+export function AuthComponent() {
```

Comment:
If you think these components will be useful in the future (and they aren't just test components that will be deleted, it might be good to give them more descriptive names.  This AuthComponent might be useful in the future if it took a children prop:
```
export function ShowIfAuthenticated(children) {
  const [{ isAuthenticated }] = useContext(context);
  return isAuthenticated && children;
};
```
.. then you could use it like this: `<ShowIfAuthenticated><p>I am auth paragraph</p></ShowIfAuthenticated>`
You don't need to do this, it's just an idea for the future.

Reply:
I got it,I just used it as a test and to make a practical example of user provider usages.
But the good naming will be helpful,
Thank You :+1:

Reply:
Sure, sounds good! Afwan.